### PR TITLE
Ka/diff tool ignore changes.w 7822674

### PR DESCRIPTION
**Pull request by kalagarraj**
Included Changes

- New rule to ignore changes in examples 
- Test cases for node changes and ignore rule
- Sorted package.json

### CODE REVIEW

File: src/diff/changes/nodeChanges.test.ts [hunk 1]
```diff
@@ -64,6 +65,55 @@ describe("Check for categorized changes in a node", () => {
...
+    expect(nodeChanges.hasIgnoredChanges()).to.be.true;
+    expect(nodeChanges.getIgnoredChangesCount()).to.equal(1);
+  });
+
+  it("returns true when there is an ignored change and a non breaking change", async () => {
+    const nodeChanges = buildNodeChanges();
+    nodeChanges.categorizedChanges[0].category = RuleCategory.IGNORED;
+    const nonBreakingChange = new CategorizedChange(
+      "r2",
+      "non-breaking-changed",
```
Comment (bredmond-sf) [inline src/diff/changes/nodeChanges.test.ts:89]:
```suggestion
      "non-breaking-change",
```

File: src/diff/changes/nodeChanges.test.ts [hunk 2]
```diff
@@ -64,6 +65,55 @@ describe("Check for categorized changes in a node", () => {
...
+    expect(nodeChanges.hasBreakingChanges()).to.be.false;
+    expect(nodeChanges.getIgnoredChangesCount()).to.equal(1);
+  });
+
+  it("returns true when there multiple ignored changes", async () => {
+    const nodeChanges = buildNodeChanges();
+    nodeChanges.categorizedChanges[0].category = RuleCategory.IGNORED;
+    const nonBreakingChange = new CategorizedChange(
+      "r2",
+      "non-breaking-changed",
```
Comment (bredmond-sf) [inline src/diff/changes/nodeChanges.test.ts:105]: same

**Comment by kalagarraj** [comment]
Closing this PR as CircleCI do not report the build status

File: src/diff/defaultRules.test.ts [hunk 3]
```diff
@@ -120,3 +124,75 @@ describe("Version change", () => {
     verifyRule(changes[0], rule);
   });
 });
+
+describe("Examples change", () => {
+  it("does not apply ignore examples rule on a change that has no apiContract:Example change type", async () => {
```
Comment (unandyala) [inline src/diff/defaultRules.test.ts:129]: apiContract:Example is a node type not a change type

File: src/diff/changes/nodeChanges.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ import { RuleCategory } from "../ruleSet";
 export class NodeChanges {
   public added: { [key: string]: unknown };
   public removed: { [key: string]: unknown };
+  public ignored: { [key: string]: unknown };
```
Comment (unandyala) [inline src/diff/changes/nodeChanges.ts:17]: what is the purpose of ignored object? I don't see it being used anywhere

File: src/diff/defaultRules.test.ts [hunk 5]
```diff
@@ -27,7 +31,7 @@ function verifyRule(nodeChanges: NodeChanges, rule: Rule): void {
       nodeChanges.added[rule.event.params.changedProperty]
     ];
   }
-  expect(categorizedChanges).to.have.length(1);
+  expect(categorizedChanges).to.have.length(countOfChanges);
```
Comment (unandyala) [inline src/diff/defaultRules.test.ts:34]: length seems to have deprecated. We should use lengthOf instead

**Pull request closed without merging**